### DISCUSSION TITLE
Pause and resume program

### DISCUSSION THREAD

**Discussion by garenor** [ekutner/home-connect-hass#206] [Community Support]
In the official Home Connect integration, there is a pause and resume service (as per documentation). But I cannot find it in this integration. Am I looking in the wrong places, or is this integration lacking this functionality? My Siemens washer has a function that pauses at a certain point. I would like a button in Home Assistant to resume the program. I can't figure out how with this integration.

I hope someone knows how to tackle this.

(Big thanks to the author(s) of this integration. I Like it!)

**Comment by rickyhoots** [comment]
Could be possible that the function is only available in the official app. There is a part in the read me here that covers this

**Comment by ekutner** [comment]
The mobile app is using a different API, however for some appliances this functionality is also available through the public API, in which case it should be supported by the integration. If the "Start" button doesn't change to "Pause" when a program is running that's probably because it's not supported, however this is not something that works for my appliance so I can't test it directly, which means there is always a chance of a bug.
To verify this is not a bug please enable debug logging (see the README) and set ```log_mode: 7 ``` in the home_connect_alt section of the configuration file. Then, make sure remote start is enable on the appliance (you should be able to start the program from the integration). While a program is running press the Home Connect Debug button, then the Home Connect Refresh button, the the Debug button again and upload the log file. Since I don't usually follow the discussion threads open this as a new issue.

**Reply by weboc69** [reply]
I also use this intergration because I feel this is more stable than the official integration.
But as the requester I also would like to use the Pause Program and Resume Program.
I think this is avalaible in the API see https://api-docs.home-connect.com/commands?#pause-program and https://api-docs.home-connect.com/commands?#resume-program
For me this can be of use the pause and resume dryer program if the solar production drops below a defined number. 
I have a Bosch dryer and a Siemens washer. 
I could test this for you if your appliances don't allow the pause and resume function. But I'm not familiar with API coding myself.
